Princeton-CDH/cdh-web
# release/3.3

- Bump django from 3.1.8 to 3.1.12 in /requirements
- Clean up unused templates; fix template tags
- Set version to 3.2 final and document changes
- Document current python dependencies
- Update npm packages via npm audit
- Bump version to 3.3-dev
- Remove landscape.io config
- Remove resources app
- Remove unused ansible and deployment configs
- Remove unused mezzanine styles and javascript
- Remove leftover template diff
- Remove outdated JSON fixtures
- Remove outdated .pdf files
- Remove resources from INSTALLED_APPS
- Add Media Central oEmbed provider (#305)
- Fix minor rst formatting problem in change log
- Set up visual regression testing with Percy
- Add migration step to visual testing workflow
- Run visual testing workflow on push
- Run django server in insecure mode to serve media in CI
- Load test site data in CI
- Update test site fixture
- Test at multiple breakpoints; test more pages
- Remove old json fixtures
- Refactor pytest fixtures into factory functions
- Add make_test_site factory function
- Revise visual test script to capture some UX
- Add basic percy config with device widths
- Use newer @percy/cli package
- Make test site fully crawlable
- Use newer percy-selenium python library
- Move test site creation to mgmt command
- Run Percy only on main/dev PRs; use latest node
- Add implicit wait so fonts load; snapshot 404 page
- Update README
- Add README badge
- Add note about required nodejs for Percy CLI
- Fix timezones and set consistent times for event fixtures
- Remove implicit waits; reference fonts via percy CSS
- Use @percy media query for percy CSS
- Make blog fixture timestamps fixed
- Fix filenames in percy font css
- Add /500 URL for testing error page
- Cleanup based on @rlskoeser/@kmcelwee review
- Visual test for 500 error page
- Re-lock js dependencies
- Add local copy of editoria11y css & js
- Add basic editoria11y integration
- Hardcode homepage URL so it works on error pages
- Ensure analytics aren't turned on in CI
- Set a static SW_VERSION in CI
- Implement automatic redirects on slug change (#354)
- Cleanup based on @rlskoeser review
- Automatically create redirects when pages are moved (#354)
- Remove local copy of editoria11y
- Adjust editoria11y inclusion and override position
- Add help text to body content embed block (#305)
- Add and document pre-commit hooks for black+isort
- Apply Black+isort to existing code
- Document styling commits to ignore for git blame
- Allow creating link pages in admin underneath homepage
- Set version to 3.3
- Update CHANGELOG for v3.3
- Rebuild docs for v3.3
- Fixup .rst formatting
- Lock python reqs for v3.3
